shader/coarse.wgsl

```diff
@@ -142,6 +145,9 @@ fn main(
     @builtin(local_invocation_id) local_id: vec3<u32>,
     @builtin(workgroup_id) wg_id: vec3<u32>,
 ) {
+    if (atomicLoad(&bump.failed) & (STAGE_BINNING | STAGE_TILE_ALLOC | STAGE_PATH_COARSE)) != 0u {
```

Comment:
Is there any advantage here to checking only whether previous stages failed (vs just doing a general non-zero check). Is it a case of branching paths in the shaders which reconverge?

As it stands, since we don't check for prior failure in `fine`, we seem to saying that if `coarse` fails, we should still run `fine` on a best effort basis? That seems fairly reasonable, although I don't think there's any guarantee that what we got out of fine is still usable?

Reply:
Ah got it, we need to dry-run all the possible allocations to get the actual final needed size. That still doesn't explain why it's not a `bump.failed & ~STAGE_COARSE` though, or why we don't check in fine.

Reply:
We could use the complement but I think the current code makes the dependencies explicit which is nice for the reader. 

Fine doesn’t contain a check because the CPU side (which isn’t implemented yet) is meant to read back the bump buffer after coarse and rerun the stages that failed until they succeed. We could also speculatively run fine and that may prove to be a reasonable choice.

Reply:
My intuition would be that reading back the bump buffer to the CPU within every frame would be extremely expensive, when we expect only a miniscule proportion of frames to require it.

But idk, GPUs are a bit unintuitive sometimes.

Reply:
Raph and I spent a great deal of time discussing that before it was implemented in piet-gpu. It ended up having a negligible impact on performance.